src/theory/bir/bir_promisingScript.sml

```diff
@@ -493,9 +491,9 @@ clstep p cid s M [] s')
    bir_get_stmt p s.bst_pc = BirStmt_Write a_e v_e xcl acq rel
  /\ (SOME l, v_addr) = bir_eval_exp_view a_e s.bst_environ s.bst_viewenv
  /\ (SOME v, v_data) = bir_eval_exp_view v_e s.bst_environ s.bst_viewenv
- /\ (xcl ==> fulfil_atomic_ok M l cid s t)
+ /\ (xcl ==> s.bst_xclb <> NONE /\ fulfil_atomic_ok M l cid (THE s.bst_xclb).xclb_time t)
```

Comment:
I'd probably write `IS_SOME s.bst_xclb` here, but that is personal taste.

Reply:
You could also avoid the `THE` by writing `case s.bst_xclb of ...`, but maybe it's overkill here.